include/smack/smack.h

```diff
@@ -49,212 +57,467 @@ void assume(int v) {
 
-  // Integer arithmetic
-  D("function {:inline} $add(p1:int, p2:int) returns (int) {p1 + p2}");
-  D("function {:inline} $sub(p1:int, p2:int) returns (int) {p1 - p2}");
-  D("function {:inline} $mul(p1:int, p2:int) returns (int) {p1 * p2}");
-  D("function {:builtin \"div\"} $sdiv(p1:int, p2:int) returns (int);");
-  D("function {:builtin \"div\"} $udiv(p1:int, p2:int) returns (int);");
-  D("function {:builtin \"rem\"} $srem(p1:int, p2:int) returns (int);");
-  D("function {:builtin \"rem\"} $urem(p1:int, p2:int) returns (int);");
-  D("function $and(p1:int, p2:int) returns (int);");
-  D("axiom $and(0,0) == 0;");
-  D("axiom $and(0,1) == 0;");
-  D("axiom $and(1,0) == 0;");
-  D("axiom $and(1,1) == 1;");
-  D("function $or(p1:int, p2:int) returns (int);");
-  D("axiom $or(0,0) == 0;");
-  D("axiom $or(0,1) == 1;");
-  D("axiom $or(1,0) == 1;");
-  D("axiom $or(1,1) == 1;");
-  D("function $xor(p1:int, p2:int) returns (int);");
-  D("axiom $xor(0,0) == 0;");
-  D("axiom $xor(0,1) == 1;");
-  D("axiom $xor(1,0) == 1;");
-  D("axiom $xor(1,1) == 0;");
-  D("function $lshr(p1:int, p2:int) returns (int);");
-  D("function $ashr(p1:int, p2:int) returns (int);");
-  D("function $shl(p1:int, p2:int) returns (int);");
-  D("function {:inline} $ult(p1:int, p2:int) returns (bool) {p1 < p2}");
-  D("function {:inline} $ugt(p1:int, p2:int) returns (bool) {p1 > p2}");
-  D("function {:inline} $ule(p1:int, p2:int) returns (bool) {p1 <= p2}");
-  D("function {:inline} $uge(p1:int, p2:int) returns (bool) {p1 >= p2}");
-  D("function {:inline} $slt(p1:int, p2:int) returns (bool) {p1 < p2}");
-  D("function {:inline} $sgt(p1:int, p2:int) returns (bool) {p1 > p2}");
-  D("function {:inline} $sle(p1:int, p2:int) returns (bool) {p1 <= p2}");
-  D("function {:inline} $sge(p1:int, p2:int) returns (bool) {p1 >= p2}");
-  D("function $nand(p1:int, p2:int) returns (int);");
-  D("function {:inline} $max(p1:int, p2:int) returns (int) {if p1 > p2 then p1 else p2}");
-  D("function {:inline} $min(p1:int, p2:int) returns (int) {if p1 > p2 then p2 else p1}");
-  D("function {:inline} $umax(p1:int, p2:int) returns (int) {if p1 > p2 then p1 else p2}");
-  D("function {:inline} $umin(p1:int, p2:int) returns (int) {if p1 > p2 then p2 else p1}");
-  D("function {:inline} $i2b(i: int) returns (bool) {i != 0}");
-  D("function {:inline} $b2i(b: bool) returns (int) {if b then 1 else 0}");
-
-  // Floating point
-  D("type float;");
-  D("function $fp(ipart:int, fpart:int, epart:int) returns (float);");
-  D("function $fadd(f1:float, f2:float) returns (float);");
-  D("function $fsub(f1:float, f2:float) returns (float);");
-  D("function $fmul(f1:float, f2:float) returns (float);");
-  D("function $fdiv(f1:float, f2:float) returns (float);");
-  D("function $frem(f1:float, f2:float) returns (float);");
+	// Integer arithmetic
+	#ifdef BITVECTOR
+	// Bitvector arithmetic
+	D("function {:bvbuiltin \"bvadd\"}   $add.i32(p1:i32, p2:i32)   returns (i32);");
```

Comment:
Never use tabs. We indent using 2 spaces. Please fix all these tab indentations.